src/jit/flowgraph.cpp

```diff
@@ -2415,25 +2415,51 @@ void Compiler::fgDfsInvPostOrderHelper(BasicBlock* block, BlockSet& visited, uns
     // Assume we haven't visited this node yet (callers ensure this).
     assert(!BlockSetOps::IsMember(this, visited, block->bbNum));
 
+    ArrayStack<DfsBlockEntry> stack(this);
+
+    DfsBlockEntry firstEntry(State_Pre, block);
```

Comment:
There certainly is an amount of personal preference in this but to me the temporary variable version doesn't seem very C++ish. While it doesn't make any difference in this case such variables prevent the use of move constructors and extend the liftetime of what would have otherwise a temporary object to the end of the scope.

Not to mention that the only reason a temporary object is needed here is the use of improvised collection classes such as `ArrayStack`. With `std::stack` it would simply be `stack.emplace(State_Pre, block);`.